napalm_nxos/nxos.py

```diff
@@ -338,8 +338,19 @@ def get_facts(self):
         pynxos_facts = self.device.facts
         final_facts = {key: value for key, value in pynxos_facts.items() if
                        key not in ['interfaces', 'uptime_string', 'vlans']}
-        final_facts['interface_list'] = pynxos_facts['interfaces']
+
+        if pynxos_facts['interfaces']:
+            final_facts['interface_list'] = pynxos_facts['interfaces']
+        else:
+            final_facts['interface_list'] = self.get_interfaces().keys()
+        
         final_facts['vendor'] = 'Cisco'
+
+        hostname_cmd = 'show hostname'
+        hostname = self.device.show(hostname_cmd).get('hostname')
```

Comment:
Perhaps I didn't put it correctly - my fault. I mean that ```fqnd``` should be the hostname only if the domain-name is not specified. If it is, should be ```<host-name>.<domain-name>```.